darts/models/forecasting/torch_forecasting_model.py

```diff
@@ -2058,6 +2063,42 @@ def _load_encoders(
 
         return new_encoders, new_add_encoders
 
+    def _check_ckpt_parameters(self, tfm_save):
+        """
+        Check that the parameters used to instantiate the new model loading the weights match those of
+        of the saved model, to return meaningful messages in case of discrepancies.
+        """
+        # parameters unrelated to the weights shape
+        skipped_params = list(
+            inspect.signature(TorchForecastingModel.__init__).parameters.keys()
+        ) + [
+            "loss_fn",
+            "torch_metrics",
+            "optimizer_cls",
+            "optimizer_kwargs",
+            "lr_scheduler_cls",
+            "lr_scheduler_kwargs",
+        ]
+        ckpt_model_params = tfm_save.model_params
+        for param_key, param_value in self.model_params.items():
```

Comment:
I'm not 100% but we might have to check the keys which are not in both dicts at the same time. If these keys are not in the `skipped_params`, we should raise an error as well (I believe `ModelMeta` only adds the positional arguments and the kwargs that were passed to `model_params`).

Also, can we capture all wrong parameters in a list, and then raise the error at the end for all wrong parameters?
Then the user can adjust all bad params at once.